kpasswd/kpasswdd.c

```diff
@@ -646,6 +646,32 @@ process (krb5_keytab keytab,
     krb5_auth_con_free(context, auth_context);
 }
 
+/*
+ * XXX this code relies on getsockname() returning a valid local
+ * address for a "connected" DGRAM socket. This is true for most, but
+ * probably not all systems. For some systems, this could be done
+ * cleaner by using the IP_RECVDSTADDR option + recvmsg().
+ */
+static int
+get_local_addr(struct sockaddr *remote, int remlen,
+	       struct sockaddr *local, socklen_t *loclen)
+{
+    int s, ret;
```

Comment:
Those are fine too.  I would expect these to be the same, no?  But you're probably right.